### PR TITLE
Update example for "Reuse prepared statement"

### PR DESCRIPTION
**Pull request by sgrossberndt**
Update the code example to prepare a statement to fetch page uids and reuse it.

[The change mentioned in the TODO](https://review.typo3.org/c/Packages/TYPO3.CMS/+/72610) has been merged years ago.

### CODE REVIEW

**Comment by sgrossberndt** [comment]
This example is valid for TYPO3 v12 and v11 as well and should be backported to those docs as well.

**Review by garvinhicking — APPROVED** [review]
Good one, thanks! Since Lina was requested I'll not merge it directly myself.

File: Documentation/ApiOverview/Database/Statement/Index.rst [hunk 1]
```diff
@@ -169,35 +169,30 @@ handy when the same query is executed over and over again with different
...
-    $statement->closeCursor(); // free the resources for this result
+        ->prepare();
+
+    $pages = [];
+    foreach ([24, 25] as $pageId) {
+        // Bind $pageId value to the first (and in this case only) positional parameter
+	    $statement->bindValue(1, $pageId, Connection::PARAM_INT);
+	    $result = $statement->executeQuery();
+        $pages[] = $result->fetchAssociative();
+        $result->free(); // free the resources for this result
```
Comment (sbuerk) [inline Documentation/ApiOverview/Database/Statement/Index.rst:194]: Note that this is only needed if not the full result set has been retrieved. For example, if the query returns 2 records but only one is fetched (single fetchAssociative()` leaving one in the result set.

That means, looping over the full result set (`while ($row = $result->fetchAssociative) {}` without breaking out, using `fetchAllAssociative()` or having a `setMaxResult(1)` on the query would make the free call obsolete. In this example however needed.

That is not related to prepared statements but also for all select/select-count queries returning a resultset.

**Review by sbuerk — APPROVED** [review]

**Review by linawolf — APPROVED** [review]
As I havent used prepared statements I will trust the two of you

**Pull request merged**